src/main/java/dk/kb/present/copyright/CopyrightAccessExtractor.java

```diff
@@ -0,0 +1,364 @@
+                     String yearStarted= yearStartedNode.item(0).getTextContent();                         
+                     coorporate.setYearStarted(yearStarted);
+                 }
+                 if (yearEndedNode.getLength() > 0){
+                     String yearEnded= yearEndedNode.item(0).getTextContent();                         
+                     coorporate.setYearEnded(yearEnded);
+                 }                                  
+                 
+                 cooperateList.add(coorporate);
+             }
+         }
+         return cooperateList;
+     }
+   
+    public static  Document createDocFromXml(String xml) throws Exception{
+
+        //System.out.println(response);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        XMLEscapeSanitiser sanitiser = new XMLEscapeSanitiser(""); //Do not replace with anything
+        String responseSanitized  =  sanitiser.apply(xml);
+
+        Document document = null;
+        try {
+            document = builder.parse(new InputSource(new StringReader(responseSanitized)));
+            document.getDocumentElement().normalize();
+        }
+        catch(Exception e) {                       
+            log.error("Invalid XML",e);                                                
+            throw new Exception("invalid xml",e);            
+
+        }
+
+        return document;
+    }
+        
+    private static String getImageLink(Document doc) {
+        NodeList identifiers = doc.getElementsByTagName("mods:identifier");
+        for (int i =0;i<identifiers.getLength();i++) {
+          
+            Element e = (Element) identifiers.item(i);        
+            String type= e.getAttribute("type");
+             
+            if ("Asset Reference".equals(type)) {
+            String ref = e.getTextContent();
+
+                ref = ref.replaceAll("cumulus-core-01:/Depot", "");
+                ref = ref.replaceAll(".tif", "");
+                                                                
+                //http://kb-images.kb.dk/?FIF=/DAMJP2/DAM/Samlingsbilleder/0000/388/116/DT005031
+                //Add last parameters such as size and format: 
+                //http://kb-images.kb.dk/?FIF=/DAMJP2/DAM/Samlingsbilleder/0000/388/116/DT005031&CVT=jpeg
+                String link = "http://kb-images.kb.dk/?FIF=/DAMJP2"+ref;            
+                return link;                
+         }
+        }
+         return null;
+     }
+    
+
+    // See documentation. Can be 3 different fields, one MUST always be there.
+    //<mods:dateCreated>1868</mods:dateCreated>  (notice no point attibute)
+    //<mods:dateCreated point="end">1900</mods:dateCreated> 
+    //<mods:dateCaptured>2014-04-04T11:52:16.000+02:00</mods:dateCaptured> 
+        
+    private static Integer getSkabelsesAar(Document doc) throws Exception{
```

Comment:
JavaDoc! As the comments above indicates that it is important that at least one date is defines, write is as JavaDoc and state in the `@throws` what happens when it is missing.